.github/workflows/docker_build_images.yml

```diff
@@ -1,22 +1,101 @@
-name: Docker Build Images
+name: Build Docker Images
 on:
+  pull_request:
   push:
     branches:
       - main
     tags:
-      - 'v[2-9][0-9]*.*' # run only on tags greater or equal to v20.0.0 where this new way of building docker image was changed
+      - '*'
 
 concurrency:
-  group: format('{0}-{1}', ${{ github.ref }}, 'Docker Build Images (v20+)')
+  group: format('{0}-{1}', ${{ github.ref }}, 'Build Docker Images')
   cancel-in-progress: true
 
 permissions: read-all
 
 jobs:
+  skip_push:
+    name: Set skip_push if we are on a Pull Request
+    runs-on: ubuntu-20.04
+    if: github.repository == 'vitessio/vitess'
+    outputs:
+      skip_push: ${{ steps.skip_push.outputs.skip_push }}
+
+    steps:
+      - name: Set skip_push
+        id: skip_push
+        run: |
+          push='false'
+          if [[ "${{github.event.pull_request}}" ==  "" ]]; then
+            push='true'
+          fi
+          echo Push='${push}'
+          echo "skip_push=${push}" >> $GITHUB_OUTPUT
+
+  build_and_push_vttestserver:
+    name: Build and push vttestserver
+    runs-on: gh-hosted-runners-16cores-1
+    if: github.repository == 'vitessio/vitess' && needs.skip_push.result == 'success'
+    needs:
+      - skip_push
+
+    strategy:
+      fail-fast: true
+      matrix:
+        branch: [ mysql80 ]
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Login to Docker Hub
+        if: needs.skip_push.outputs.skip_push == 'true'
```

Comment:
Why would we need to login to docker hub if we're NOT pushing?

Reply:
That's a really good point, the naming of `skip_push` was misleading, I renamed it via e99c1a7622